test/CMakeLists.txt

```diff
@@ -27,3 +27,6 @@ target_link_libraries(dynamic-override PUBLIC mimalloc)
 
 add_executable(dynamic-override-cxx  main-override.cpp)
 target_link_libraries(dynamic-override-cxx PUBLIC mimalloc)
+
+add_executable(main-c  main.c)
```

Comment:
Would you mind if you split this change against `main-c` into another pull request? So that, we can concentrate on `PATH_MAX` discussions and review.

Reply:
Yep. (but I wouldn't open a separate PR for that, so I just drop that patch)